src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -601,7 +601,7 @@ async Task OnStreamCanceled(InvocationRequest irq)
             var readers = default(Dictionary<string, object>);
 
             CheckDisposed();
-            var connectionState = await _state.WaitForActiveConnectionAsync(nameof(StreamAsChannelCoreAsync));
+            var connectionState = await _state.WaitForActiveConnectionAsync(nameof(StreamAsChannelCoreAsync), token: cancellationToken);
```

Comment:
Can we remove `token = default` from all of our internal methods that wait for the semaphore like WaitForActiveConnectionAsync, SendWithLock, WaitConnectionLockAsync, etc...?

I know WaitConnectionLockAsync in particular has a lot of usages where we don't want to use a cancellation token for close/reconnect logic, but I like being explicit.

Reply:
Fixed